h/api/views.py

```diff
@@ -201,20 +202,37 @@ def read_jsonld(annotation, request):
     return presenter.asdict()
 
 
-@api_config(route_name='api.annotation', request_method='PUT', permission='update')
+@api_config(route_name='api.annotation',
+            request_method='PUT',
+            permission='update')
 def update(annotation, request):
     """Update the specified annotation with data from the PUT payload."""
-    schema = schemas.LegacyUpdateAnnotationSchema(request,
-                                                  annotation=annotation)
-    appstruct = schema.validate(_json_payload(request))
-    annotation = storage.legacy_update_annotation(request,
-                                                  annotation.id,
-                                                  appstruct)
+    json_payload = _json_payload(request)
 
-    _publish_annotation_event(request, annotation, 'update')
+    legacy_annotation = elastic.Annotation.fetch(annotation.id)
 
-    presenter = AnnotationJSONPresenter(request, annotation)
-    return presenter.asdict()
+    # Validate the annotation for, and update the annotation in, PostgreSQL.
+    if request.feature('postgres'):
+        schema = schemas.UpdateAnnotationSchema(request, annotation=annotation)
+        appstruct = schema.validate(copy.deepcopy(json_payload))
+        annotation = storage.update_annotation(request.db,
+                                               annotation.id,
+                                               appstruct)
```

Comment:
Hmm. I tried this:

1. Create an annotation with postgres on
2. Delete the annotation from postgres with psql.
3. Annotation will still be showing in the client so you can edit it. Try to update the annotation.

This simulates what would happen if we somehow had an annotation in the (new) Es index but not in Pg and someone tried to update it. What happens is that `h.api.resources.AnnotationFactory.__getitem__()` calls `storage.fetch_annotation()` which returns `None` (because it's not in the db) and then we 404. It doesn't get to the view, schema or storage code for updating the annotation.

Do we want to do something about this? Like on an annotation update attempt, if the annotation isn't found in Pg, attempt to fetch it from the legacy Es and create it in Pg?

I think maybe we just want to keep it simple and not handle this case on the update path